hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsUriQueryBuilder.java

```diff
@@ -40,6 +41,10 @@ public void addQuery(final String name, final String value) {
     }
   }
 
+  public void setSASToken(final String sasToken) {
+    this.sasToken = sasToken;
```

Comment:
is it a requirement that the token is unset? if so, add a Preconditions.checkState. if not, then not

similarly: any checks on token (not null/empty)

Reply:
Didn't modify as the QueryBuilder will be used in a state where SasToken need not be mandatorily set. If set, will be used by ToString().